app/src/main/java/com/shopapp/ShopApplication.kt

```diff
@@ -33,7 +35,7 @@ open class ShopApplication : Application() {
         val api = MagentoApi(this, "http://10.14.14.29/")
         val dao = DaoImpl(this)
 
-        appComponent = buildAppComponent(api, dao)
+        appComponent = buildAppComponent(api, dao, api.getConfig())
```

Comment:
Do we really need to pass `api.getConfig()` separately if we already have `api` as the first argument?